tasks/winbuildscripts/Generate-Chocolatey-Package.ps1

```diff
@@ -64,19 +64,15 @@ Write-Host "Generating Chocolatey $installMethod package version $agentVersion i
 Write-Host ("Downloading {0}" -f $url)
 $statusCode = -1
 try {
-    $req = [System.Net.WebRequest]::Create($url)
-    $rep = $req.GetResponse()
-    $statusCode = $rep.StatusCode
+    $tempMsi = "$(Get-Location)\ddagent.msi"
+    Remove-Item $tempMsi -ErrorAction Ignore
+    (New-Object net.webclient).Downloadfile($url, $tempMsi)
+    $checksum = (Get-FileHash $tempMsi -Algorithm SHA256).Hash
+    Remove-Item $tempMsi
 }
-catch [System.Net.WebException] {
-    if ($_.Exception.Status -eq "ProtocolError") {
-        $statusCode = [int]$_.Exception.Response.StatusCode
-    }
-}
-Write-Host $statusCode
-
-if ($statusCode -ne 200) {
-    Write-Warning "Package $($url) doesn't exists yet, make sure it exists before publishing the Chocolatey package !"
+catch {
+    Write-Host "Could not generate checksum for package $($url): $($_)"
```

Comment:
Shouldn't we display the exception's text to be able to discriminate between a checksum error and a checksum error?

Reply:
woops, I missed that this was a backport. Then please ignore